### PR TITLE
avoid failure of plot_feature_outlier_image

### PR DESCRIPTION
**Pull request by signupatgmx**
I call the function this way:
```py
plot_feature_outlier_image(preds,  # output of an prediction
                                   ds, # my set of images
                                   X_recon=recon, # reconstructed set of images
                                   max_instances=5,       # max nb of instances to display
                                   n_channels=3,          # number of channels of the images
                                   figsize=(20,10),
                                   outliers_only=True)    # only show outlier predictions. this function fails when no outlier is detected and parameter set to True
```
In case there is no outlieres detected and the parameter outliers_only is set to True, then the function fails, because it calls plt.subplots(nrows=0, ....). 0 rows throws the error.

### CODE REVIEW

**Review by mauicv — CHANGES_REQUESTED** [review]
Hey @signupatgmx,
Thank you very much for opening this PR! I think the change makes sense, one minor ask would be to add a warning if the `outliers_only and n_instances == 0` condition is met. If the function returns without warning the user then it's not entirely clear what happened. Lmk if you are happy to do this otherwise I'm happy to pick it up!
Thanks regardless

**Comment by signupatgmx** [comment]
Hallo @mauicv, as I'm not customized with your coding standards, I'd suggest you add a suitable warning output. I'm glad to have found your great library.
Kind regards.

File: alibi_detect/utils/visualize.py [hunk 1]
```diff
@@ -77,6 +78,11 @@ def plot_feature_outlier_image(od_preds: Dict,
         instance_ids = list(range(len(od_preds['data']['is_outlier'])))
     n_instances = min(max_instances, len(instance_ids))
     instance_ids = instance_ids[:n_instances]
+
+    if outliers_only and n_instances == 0:
+        warnings.warn('No outliers found!', UserWarning, stacklevel=3)
```
Comment (ascillitoe) [inline alibi_detect/utils/visualize.py:83]: @mauicv  is the warning swallowed without `stacklevel=3`?

Comment (mauicv) [reply alibi_detect/utils/visualize.py:83]: Nope, I've removed it!

**Review by ascillitoe — APPROVED** [review]
LGTM!

**Pull request merged**